lib/semantics/check-call.cpp

```diff
@@ -610,11 +611,26 @@ static void CheckExplicitInterfaceArg(evaluate::ActualArgument &arg,
                 messages.Say(
                     "Actual argument is not a variable or typed expression"_err_en_US);
               }
-            } else if (!object.type.type().IsAssumedType()) {
+            } else {
               const Symbol &assumed{DEREF(arg.GetAssumedTypeDummy())};
-              messages.Say(
-                  "Assumed-type TYPE(*) '%s' may be associated only with an assumed-TYPE(*) %s"_err_en_US,
-                  assumed.name(), dummyName);
+              if (!object.type.type().IsAssumedType()) {
+                messages.Say(
+                    "Assumed-type TYPE(*) '%s' may be associated only with an"
+                    " assumed-TYPE(*) %s"_err_en_US,
```

Comment:
The second part of the message looks garbled.

Reply:
It does look a little wacky.  But I didn't change the message, I only changed
the string formatting to fit into 80 columns.  Here's a sample of how an actual message appears:
```                           
x.f90:4:3: error: Assumed-type TYPE(*) 'var' may be associated only with an assumed-TYPE(*) dummy argument 'arg='
```
If you think the message should be changed, let me know.

Reply:
Can you please just fix the message?

Reply:
Will do.  The obvious way to clarify this message is to add the phrase `actual argument` to the start of the message.  This would make the message read as follows:
```
Assumed-type TYPE(*) actual argument 'var' may be  associated only with an assumed-TYPE(*) dummy argument 'arg='
```
But it also seems wrong to me that the name of the dummy argument as an equal sign appended to the end of it.  This equal sign ends up appearing in many error messages associated with dummy arguments, so I assumed that it must make some kind of sense.  But it doesn't make sense to me.  Do you have any insight on why the message contains the equal sign?

Reply:
I give up.  Leave it alone.  I'll take care of it.